Objective-C/CBLDatabase.mm

```diff
@@ -627,18 +658,18 @@ - (nullable CBLDocument*) documentWithID: (NSString*)documentID
                                mustExist: (bool)mustExist
                                    error: (NSError**)outError
 {
-    [self mustBeOpen];
-    
-    return [[CBLDocument alloc] initWithDatabase: self
-                                      documentID: documentID
-                                       mustExist: mustExist
-                                           error: outError];
+    @synchronized (self) {
```

Comment:
Yes, it could make concurrent read slow. What is your suggestion on this?

Side note: I'm trying not to make complicated locks until we really have the real use cases that have performance issue. But I'm open for discussion.